### PR TITLE
feat: ListUsers dispatch throttling

### PR DESCRIPTION
**Pull request by willvedd**
## Description
This PR adds dispatch throttling to the ListUsers API. It is very similar to the implementation of #1502 which introduced dispatch throttling to ListObjects. 

## References
- Related PR: #1502

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Review by poovamraj — DISMISSED** [review]

File: pkg/server/server.go [hunk 1]
```diff
@@ -428,6 +434,44 @@ func WithListObjectsDispatchThrottlingMaxThreshold(maxThreshold uint32) OpenFGAS
...
+// which diminishes the value of the throttling.
+func WithListUsersDispatchThrottlingFrequency(frequency time.Duration) OpenFGAServiceV1Option {
+	return func(s *Server) {
+		s.listUsersDispatchThrottlingFrequency = frequency
+	}
+}
+
+// WithListUsersDispatchThrottlingThreshold define the number of dispatches to be throttled
+// for List Objects requests.
+func WithListUsersDispatchThrottlingThreshold(threshold uint32) OpenFGAServiceV1Option {
```
Comment (miparnisari) [inline pkg/server/server.go:461]:
```suggestion
// WithListUsersDispatchThrottlingThreshold define the number of dispatches to be throttled
// for ListUsers requests.
func WithListUsersDispatchThrottlingThreshold(threshold uint32) OpenFGAServiceV1Option {
```

File: pkg/server/server.go [hunk 2]
```diff
@@ -444,6 +451,44 @@ func WithListObjectsDispatchThrottlingMaxThreshold(maxThreshold uint32) OpenFGAS
 	}
 }
 
+// WithListUsersDispatchThrottlingEnabled sets whether dispatch throttling is enabled for List Objects requests.
```
Comment (miparnisari) [inline pkg/server/server.go:454]: nit, here and below
```suggestion
// WithListUsersDispatchThrottlingEnabled sets whether dispatch throttling is enabled for List Users requests.
```

File: pkg/server/server.go [hunk 3]
```diff
@@ -549,6 +603,16 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 		s.listObjectsDispatchThrottler = throttler.NewConstantRateThrottler(s.listObjectsDispatchThrottlingFrequency, "list_objects_dispatch_throttle")
 	}
 
+	if s.listUsersDispatchThrottlingEnabled {
+		s.logger.Info("Enabling ListUsers dispatch throttling",
```
Comment (miparnisari) [inline pkg/server/server.go:607]: nit, you don't need this log, it should be logged automatically on server startup (https://github.com/openfga/openfga/pull/1609)

**Review by miparnisari — APPROVED** [review]

File: pkg/server/server.go (hunk 3 above)
Comment (elbuo8) [reply pkg/server/server.go:607]: Yes, this would just add noise. Please drop.

Comment (miparnisari) [reply pkg/server/server.go:607]: Ah wait, there is one caveat, the line that prints all the configs (https://github.com/openfga/openfga/blob/f10d8e5df8e3d0ecfcd91a4e1a772dab32d8d360/cmd/run/run.go#L619-L625) doesn't run if you import OpenFGA as a library 😭

**Review by miparnisari — APPROVED** [review]

**Review by poovamraj — APPROVED** [review]

**Pull request merged**